### ISSUE TITLE
Mystical Index Transfer API Issue

### ISSUE THREAD

**Issue by NickMesser** [Sinytra/Connector#467]
### Describe the bug

When trying to use my library block it crashes the server. 

### Steps to reproduce

1. Spawn Book Of Storage. Add item to book of storage by shift right clicking any block and dropping the block. 
2. Spawn library and put a hopper under it. Try to open inventory. 

If there is a way to fix this on my side please let me know. The library uses some custom logic to hopper items out of the books instead of the book themselves using the transfer api. Other than this, the mod works fine!

...
[crash-2023-10-31_20.44.57-server.txt](https://github.com/Sinytra/Connector/files/13222318/crash-2023-10-31_20.44.57-server.txt)

**Comment by Su5eD** [comment]
Besides not making FFAPI crash when nothing is extracted, I've tried out using the Book of Storage on the latest version of Mystical Index, yet it doesn't seem to work on neither Forge nor Fabric.

On Fabric loader, I've got Mystical Index 1.20.1-1.3 and MidnightLib 1.4.1 installed. Shift-right-clicking a block seems to write it to the book, but it's not removed from the world. Does this happen on your side, too?
